src/test/java/apoc/algo/WeaklyConnectedComponentsTest.java

```diff
@@ -0,0 +1,126 @@
+            "CREATE (:Node:Record { name : 'I' })  "+
+            "WITH count(*) as dummy " +
+            "CREATE (:Node:Record { name : 'J' })  "+
+            "WITH count(*) as dummy " +
+            "CREATE (:Node:Record { name : 'K' })  "+
+            "WITH count(*) as dummy " +
+            "CREATE (:Node:Record { name : 'L' })  "+
+            "WITH count(*) as dummy " +
+            "CREATE (:Node:Record { name : 'M' })  "+
+            "WITH count(*) as dummy " +
+            "CREATE (:Node:Record { name : 'N' })  "+
+            "WITH count(*) as dummy " +
+            "CREATE (:Node:Record { name : 'Z' })  "+
+            "WITH count(*) as dummy " +
+            "MATCH (a:Node),(b:Node),(c:Node)  "+
+            "WHERE a.name = 'C' AND b.name = 'D' AND c.name = 'E'  "+
+            "CREATE (a)-[:LINK]->(b)-[:LINK]->(c)-[:LINK]->(a)  "+   
+            "WITH count(*) as dummy " +
+            "MATCH (a:Node),(b:Node)  "+
+            "WHERE a.name = 'F' AND b.name = 'G'  "+ 
+            "CREATE (a)-[:LINK]->(b)-[:LINK]->(a)  "+
+            "WITH count(*) as dummy " +
+            "MATCH (a:Node),(b:Node)  "+
+            "WHERE a.name = 'H' AND b.name = 'Z'   "+
+            "CREATE (a)-[:LINK]->(b)-[:LINK]->(a)  "+
+            "WITH count(*) as dummy " +
+            "MATCH (a:Node),(b:Node)  "+
+            "WHERE a.name = 'I' AND b.name = 'Z'  "+ 
+            "CREATE (a)-[:LINK]->(b)  "+
+            "WITH count(*) as dummy " +
+            "MATCH (a:Node),(b:Node)  "+
+            "WHERE a.name = 'J' AND b.name = 'Z'   "+
+            "CREATE (a)-[:LINK]->(b)  "+
+            "WITH count(*) as dummy " +
+            "MATCH (a:Node),(b:Node)  "+
+            "WHERE a.name = 'K' AND b.name = 'Z'   "+
+            "CREATE (a)-[:LINK]->(b)  "+
+            "WITH count(*) as dummy " +
+            "MATCH (a:Node),(b:Node)  "+
+            "WHERE a.name = 'L' AND b.name = 'Z'   "+
+            "CREATE (a)-[:LINK]->(b)  "+
+            "WITH count(*) as dummy " +
+            "MATCH (a:Node),(b:Node)  "+
+            "WHERE a.name = 'M' AND b.name = 'Z'   "+
+            "CREATE (a)-[:LINK]->(b)  "+
+            "WITH count(*) as dummy " +
+            "MATCH (a:Node),(b:Node)  "+
+            "WHERE a.name = 'N' AND b.name = 'Z'   "+
+            "CREATE (a)-[:LINK]->(b)";
+
+	@BeforeClass
+	public static void setUp() throws Exception {
+		db = new TestGraphDatabaseFactory().newImpermanentDatabase();
+		TestUtil.registerProcedure(db, WeaklyConnectedComponents.class);
+	}
+
+	@AfterClass
+	public static void tearDown() {
+		db.shutdown();
+	}
+
+	// ==========================================================================================
+
+	@Test
+	public void shouldReturnExpectedResultCountWhenUsingWeaklyConnectedWithEmptyDb() {
+		assertExpectedResult(0L, "CALL apoc.algo.wcc()");
+	}
+
+	@Test
+	public void shouldReturnExpectedResultCountWhenUsingWeaklyConnected() {
+		db.execute(CC_GRAPH).close();
```

Comment:
this should happen in setup, as the setup is static, you probably want to make the setup non-static here